native-sql-engine/cpp/src/codegen/arrow_compute/ext/actions_impl.cc

```diff
@@ -1977,6 +2316,183 @@ class AvgAction : public ActionBase {
   uint64_t length_ = 0;
 };
 
+// Decimal
+template <typename DataType, typename CType, typename ResDataType, typename ResCType>
+class AvgAction<DataType, CType, ResDataType, ResCType,
```

Comment:
is AVG on decimal column a valid case? seems on Spark it will cast decimal to long first

Reply:
It looks like sum could be decimal, count is long type. And average could be decimal type.

```
  private lazy val resultType = child.dataType match {
    case DecimalType.Fixed(p, s) =>
      DecimalType.bounded(p + 4, s + 4)
    case _ => DoubleType
  }

  private lazy val sumDataType = child.dataType match {
    case _ @ DecimalType.Fixed(p, s) => DecimalType.bounded(p + 10, s)
    case _ => DoubleType
  }

  private lazy val sum = AttributeReference("sum", sumDataType)()
  private lazy val count = AttributeReference("count", LongType)()
```